src/main/java/org/opentripplanner/common/model/T3.java

```diff
@@ -0,0 +1,10 @@
+package org.opentripplanner.common.model;
+
+import java.io.Serializable;
+
+/**
+ * An ordered object of three (potentially different types)
+ */
+public record T3<E1, E2, E3>(E1 first, E2 second, E3 third) implements Serializable {
```

Comment:
Again, replace this with a named record: with Java 17 they are very easy to create and much less confusing to use.

Reply:
I actually ended up deleting T3 as it was only used in P3 for inheritance and since that was converted into a record, there was no use for this class anymore.